eZ/Publish/Core/Repository/Tests/Service/Mock/UserPreferenceTest.php

```diff
@@ -0,0 +1,220 @@
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\Tests\Service\Mock;
+
+use Exception;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\Core\Repository\Tests\Service\Mock\Base as BaseServiceMockTest;
+use eZ\Publish\Core\Repository\UserPreferenceService;
+use eZ\Publish\Core\Repository\Values\User\UserReference;
+use eZ\Publish\SPI\Persistence\UserPreference\UserPreference;
+use eZ\Publish\SPI\Persistence\UserPreference\UserPreferenceSetStruct;
+use eZ\Publish\API\Repository\Values\UserPreference\UserPreferenceSetStruct as APIUserPreferenceSetStruct;
+use eZ\Publish\API\Repository\Values\UserPreference\UserPreference as APIUserPreference;
+
+class UserPreferenceTest extends BaseServiceMockTest
+{
+    const CURRENT_USER_ID = 14;
+    const USER_PREFERENCE_NAME = 'setting';
+    const USER_PREFERENCE_VALUE = 'value';
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\UserPreference\Handler|\PHPUnit\Framework\MockObject\MockObject
+     */
+    private $userSPIPreferenceHandler;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->userSPIPreferenceHandler = $this->getPersistenceMockHandler('UserPreference\\Handler');
+        $permissionResolverMock = $this->createMock(PermissionResolver::class);
+        $permissionResolverMock
+            ->expects($this->atLeastOnce())
+            ->method('getCurrentUserReference')
+            ->willReturn(new UserReference(self::CURRENT_USER_ID));
+        $repository = $this->getRepositoryMock();
+        $repository
+            ->expects($this->atLeastOnce())
+            ->method('getPermissionResolver')
+            ->willReturn($permissionResolverMock);
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Repository\UserPreferenceService::setUserPreference()
+     */
+    public function testSetUserPreference()
+    {
+        $apiUserPreferenceSetStruct = new APIUserPreferenceSetStruct([
+            'name' => 'setting',
+            'value' => 'value',
+        ]);
+
+        $this->assertTransactionIsCommitted(function () {
+            $this->userSPIPreferenceHandler
+                ->expects($this->once())
+                ->method('setUserPreference')
+                ->willReturnCallback(function (UserPreferenceSetStruct $setStruct) {
+                    $this->assertEquals(self::USER_PREFERENCE_NAME, $setStruct->name);
+                    $this->assertEquals(self::USER_PREFERENCE_VALUE, $setStruct->value);
+                    $this->assertEquals(self::CURRENT_USER_ID, $setStruct->userId);
+
+                    return new UserPreference();
+                });
+        });
+
+        $this->createAPIUserPreferenceService()->setUserPreference([$apiUserPreferenceSetStruct]);
+    }
+
+    /**
+     * @covers \eZ\Publish\Core\Repository\UserPreferenceService::setUserPreference
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
```

Comment:
Hmm, I think you're the one who showed us PHPUnit's `$this->expectException()`, why go back to old standard? ;)

Reply:
Because of respect to consistency with other tests :)

Reply:
Consistency is tricky. We started some time ago to introduce ->expectException to kernel tests as well, but you have a good point. 
Let's keep it here as is and decide later on how to approach such improvements.